test/test_ssa_direct_qmc.cc

```diff
@@ -0,0 +1,101 @@
+#include <string>
+#include <cstring>
+#include <cstddef>
+#include <cmath>
+#include <iostream>
+#include <algorithm>
+#include <numeric>
+#include <vector>
+#include <random>
+#include <stdexcept>
+
+#include "rdmini/vandercorput.h"
+#include "rdmini/ssa_direct.h"
+
+#include "gtest/gtest.h"
+
+//class ssa_distribution : public ::testing:Test {
+//protected: 
+//    virtual void SetUp () {
+
+
+namespace KHineq {
+    // Compute theoretical bound on the discrepancy of the VdC sequence:
+    // see P.Kritzer, "A new upper bound on the star discrepancy of 
+    // (0,1)-sequences", Electronic journal of combinatorial number theory 2005:
+    //      Dstar(x1,...,xN) <= f_b * logN/N + c_b * 1/N
+    // This value is then used in Koksma-Hlawka inequality.
+    constexpr size_t base = 10;
+    constexpr double a_b  = (base%2)?((base-1.)/4.):((base*base)/(4.*(base+1.)));
+    constexpr double f_b  = a_b/std::log(base);
+    constexpr double c_b  = (2.0>1.0+1.0/base+a_b)?(2.0):(1.0+1.0/base+a_b);
+    
+    template <typename Fun>
+    void kh_test(const rdmini::ssa_direct<size_t,double>& ssa, const Fun& f, double V_f, double exact_mu, std::size_t n_events) {
+        rdmini::counting_generator Rlin;
+        rdmini::vdc_uniform_real_distribution<double> U_vdc(0.,1.);
+        double approx_mu =0.;
+        double err_bound_theory = 0.;
+        std::ostringstream os;
+        for (std::size_t N=0; N<n_events; ++N) {
+            approx_mu       = ( approx_mu*N + f(ssa.inverse_cdf(U_vdc(Rlin))) )/(N+1);
+            err_bound_theory  =  V_f*(f_b*std::log(N+1)/(N+1)+c_b/(N+1));
+            if (std::abs(approx_mu-exact_mu) > err_bound_theory) {
+                os << "After " << N << " iterations:"
```

Comment:
A bit picky, I know! But the `N` here should be `N+1`. Rather than have all these N+1 terms, perhaps it'd be cleaner to have the loop start from 1?